mimic/test/test_core.py

```diff
@@ -92,3 +102,243 @@ def cleanup():
+        core = helper.core
+
+        self.assertIsNone(
+            core.service_with_region(
+                u"EXTERNAL",
+                u"some-region-name",
+                u"http://some/random/prefix"
+            )
+        )
+
+    def test_service_with_region_internal(self):
+        """
+        Validate that an external service does not
+        provide an internal service resource
+        """
+        iapi = ExampleAPI()
+        self.assertIsNotNone(iapi)
+        helper = APIMockHelper(self, [iapi])
+        core = helper.core
+
+        service_id = None
+        for a_service_id in core._uuid_to_api_internal:
+            if core._uuid_to_api_internal[a_service_id] == iapi:
+                service_id = a_service_id
+
+        resource = core.service_with_region(
+            u"ORD",
+            service_id,
+            u"http://some/random/prefix"
+        )
+        self.assertTrue(
+            IResource.providedBy(resource)
+        )
+
+    def test_uri_for_service(self):
+        """
+        Validate that the URI returned by uri_for_service
+        returns the appropriate base URI
+        """
+        iapi = ExampleAPI()
+        self.assertIsNotNone(iapi)
+        helper = APIMockHelper(self, [iapi])
+        core = helper.core
+
+        service_id = None
+        for a_service_id in core._uuid_to_api_internal:
+            if core._uuid_to_api_internal[a_service_id] == iapi:
+                service_id = a_service_id
+
+        base_uri = "http://some/random/prefix"
+
+        uri = core.uri_for_service(
+            u"ORD",
+            service_id,
+            base_uri
+        )
+        self.assertTrue(
+            uri.startswith(base_uri + '/')
+        )
+
+    def test_entries_for_tenant_external(self):
+        """
+        Validate that the external API shows up in the service catalog for
+        a given tenant.
+        """
+        eeapi = make_example_external_api(
+            name=self.eeapi_name,
+            set_enabled=True
+        )
+
+        core = MimicCore(Clock(), [eeapi])
+
+        prefix_map = {}
+        base_uri = "http://some/random/prefix"
+        catalog_entries = [
+            entry
+            for entry in core.entries_for_tenant(
+                'some-tenant',
+                prefix_map,
+                base_uri
+            )
+        ]
+
+        self.assertEqual(len(catalog_entries), 1)
+
+    def test_entries_for_tenant_internal(self):
+        """
+        Validate that the internal API shows up in the service catalog for
+        a given tenant.
+        """
+        core = MimicCore(Clock(), [ExampleAPI()])
+
+        prefix_map = {}
+        base_uri = "http://some/random/prefix"
+        catalog_entries = [
+            entry
+            for entry in core.entries_for_tenant(
+                'some-tenant',
+                prefix_map,
+                base_uri
+            )
+        ]
+
+        self.assertEqual(len(catalog_entries), 1)
```

Comment:
Is checking the length of the catalog entries a sufficiently strict test? The concern I have is that this only validates that there is a plugin in the catalog, not that an instance of ExampleAPI is provided by the catalog.